lib/charms/storage_libs/v0/cephfs_interfaces.py

```diff
@@ -0,0 +1,375 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Library to manage integrations between CephFS share providers and consumers.
+
+This library contains the CephFSProvides and CephFSRequires classes for managing an
+integration between a CephFS server operator and a CephFS client operator.
+
+## CephFSRequires (CephFS Client)
+
+This class provides a uniform interface for charms that need to mount, unmount,
+or request CephFS shares, and convenience methods for consuming data sent by a
+CephFS server charm.
+
+### Defined events
+
+- `server_connected`: Event emitted when the CephFS client is connected to the CephFS server.
+    Here is where CephFS clients will commonly request the CephFS share they need created.
+- `mount_share`: Event emitted when the CephFS share is ready to be mounted.
+- `umount_share`: Event emitted when the CephFS share is ready or needs to be unmounted.
+
+> __Note:__ This charm library only supports a 1-on-1 relation between a CephFS server and a CephFS client.
+> This is to prevent the CephFS client from having to manage and request multiple CephFS shares,
+> and ensure that CephFS clients are creating unique mount points.
+
+## CephFSProvides (CephFS Server)
+
+This library provides a uniform interface for charms that need to process CephFS share
+requests, and convenience methods for consuming data sent by a CephFS client charm.
+
+### Defined events
+
+- `share_requested`: Event emitted when the CephFS client requests a CephFS share.
+
+> __Note:__ It is the responsibility of the CephFS Provider charm to provide
+> the implementation for creating a new CephFS share. CephFSProvides just provides
+> the interface for the integration.
+"""
+
+import json
+import logging
+from dataclasses import dataclass, asdict
+from typing import Dict, List, Optional, Set, Union, Iterable
+
+from ops.charm import (
+    CharmBase,
+    CharmEvents,
+    RelationChangedEvent,
+    RelationDepartedEvent,
+    RelationEvent,
+    RelationJoinedEvent,
+)
+from ops.framework import EventSource, Object
+from ops.model import Relation
+
+# The unique Charmhub library identifier, never change it
+LIBID = "UNPUBLISHED"
```

Comment:
This suggestion will apply the `LIBID` I created for cephfs_interfaces. Once it's applied, we can uploaded this library to Charmhub:

```suggestion
LIBID = "874169fd0b874bbeb616941ada231d99"
```